### PR TITLE
Add jdk_restricted_security test

### PR DESCRIPTION
**Pull request by LongyuZhang**
Add jdk_restricted_security test

Issue: Runtimes_Automation 119

### CODE REVIEW

**Comment by LongyuZhang** [comment]
Test link:
- hyc_grinder 45206

**Comment by taoliult** [comment]
Another question: Does the directory structure in this PR https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/871 look good?

File: openjdk/playlist.xml [hunk 1]
```diff
@@ -1657,6 +1657,35 @@
...
+		<command>$(JAVA_COMMAND) -Xmx512m -jar $(Q)$(TEST_RESROOT)$(D)jtreg$(D)lib$(D)jtreg.jar$(Q) \
+	$(JTREG_BASIC_OPTIONS) $(JDK_NATIVE_OPTIONS) -vmoptions:$(Q)-Xmx512m $(JVM_OPTIONS) $(VMOPTION_HEADLESS)$(Q) \
+	$(TIMEOUT_HANDLER) \
+	-w $(Q)$(REPORTDIR)$(D)work$(Q) \
+	-r $(Q)$(REPORTDIR)$(D)report$(Q) \
+	-jdk:$(Q)$(TEST_JDK_HOME)$(Q) \
+	$(Q)$(OPENJDK_DIR)$(D)closed$(D)test$(D)jdk$(D)openj9$(D)internal$(D)security$(Q); \
+	$(TEST_STATUS)</command>
+		<features>
+			<feature>FIPS140_3_OpenJCEPlusFIPS:required</feature>
```
[the remark this answers is not in the document]
Comment (LongyuZhang) [reply openjdk/playlist.xml:1675]: Is this test `Semeru` specific, or it also works for `Eclipse`?
Also, if this test is not fips specific, then we can only run it in normal mode then. FYI @llxia

File: openjdk/playlist.xml [hunk 2]
```diff
@@ -1657,6 +1657,35 @@
 			<feature>FIPS140_3_OpenJCEPlusFIPS.FIPS140-3:nonapplicable</feature>
 		</features>
 	</test>
+	<test>
+		<testCaseName>jdk_restricted_security_j9</testCaseName>
+		<disable>
+			<comment>Only applicable on aix, pliunx, xlinux, wins64 and zlinux atm</comment>
+			<platform>^((?!(ppc64_aix|ppc64le_linux|x86-64_linux|x86-64_windows|s390x_linux)).)*$</platform>
```
[the remark this answers is not in the document]
Comment (LongyuZhang) [reply openjdk/playlist.xml:1664]: Does it apply for Z/os? If not, we can just exclude Z/os then.

File: openjdk/playlist.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (llxia) [reply openjdk/playlist.xml:1675]: if this test is not fips specific, please use `nonapplicable`. It means the test only runs in non fips testing (regular testing). See https://github.com/adoptium/aqa-tests/blob/604a0526824f170132ec66486b17cc0aa6e997c4/openjdk/playlist.xml#L566-L570

[the remark this answers is not in the document]
Comment (LongyuZhang) [reply openjdk/playlist.xml:1675]: Updated the `features`, thanks.

[the remark this answers is not in the document]
Comment (taoliult) [reply openjdk/playlist.xml:1675]: > Is this test `Semeru` specific, or it also works for `Eclipse`? Also, if this test is not fips specific, then we can only run it in normal mode then. FYI @llxia

This is `Semeru` specific.

File: openjdk/playlist.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (taoliult) [reply openjdk/playlist.xml:1664]: No, it doesn't apply for Z/os.

**Comment by LongyuZhang** [comment]
Test links (hyc_grinder):
- (pass) xlinux:  45446
- (pass) zlinux: 45458
- (pass) plinux: 45459
- aix JDK11: 45455
    - `TestProperties passed, but TestProviders failed`
- xmac JDK8: 45456
    - `Error: Cannot find file: /Users/jenkins/workspace/Grinder/aqa-tests/TKG/../openjdk/openjdk-jdk/closed/test/jdk/openj9/internal/security `, may need to modify the path for it.
- amac JDK11: 45457
    - `TestProperties passed, but TestProviders failed`
- win: JDK11: 45460
    -  `TestProperties passed, but TestProviders failed`

**Comment by taoliult** [comment]
@LongyuZhang For those failed ones, do you have the job link for it?

**Comment by llxia** [comment]
> For those failed ones, do you have the job link for it?


@taoliult please use Grinder job in hyc + job number in https://github.com/adoptium/aqa-tests/pull/5782#issuecomment-2515710226

**Comment by taoliult** [comment]
@LongyuZhang @llxia 

For `aix JDK11: 45455`, `amac JDK11: 45457` and `win: JDK11: 45460`, working on the fix, because for provider SunJSSE, its fully qualified name in JDK11 is `com.sun.net.ssl.internal.ssl.Provider`.

For `xmac JDK8: 45456m`. The error is “`[2024-12-03T22:23:00.452Z] Error: Cannot find file: /Users/jenkins/workspace/Grinder/aqa-tests/TKG/../openjdk/openjdk-jdk/closed/test/jdk/openj9/internal/security`”. 

The test cases in JDK8 are in the directory “`https://github.com/ibmruntimes/openj9-openjdk-jdk8/tree/openj9/closed/adds/jdk/test/jdk/openj9/internal/security`” The “`/closed/adds/jdk/test/jdk/openj9/internal/security`” directory, same directory as the `closed src` folder, not the “`/closed/test/jdk/openj9/internal/security`”.

Do I need to change the test case codes directory in JDK8 to “`/closed/test/jdk/openj9/internal/security`”?

**Comment by taoliult** [comment]
@LongyuZhang @llxia 

I am using my personal build and personal repo for testing `TestProviders failed `fix. But the test is disabled due to "`only applicable for Semeru atm`". Can't this test be used for nightly build or personal build?

```
18:55:07  ===============================================
18:55:07  Running test jdk_restricted_security_j9_0 ...
18:55:07  ===============================================
18:55:07  jdk_restricted_security_j9_0 Start Time: Wed Dec  4 18:55:07 2024 Epoch Time (ms): 1733356507576
18:55:07  jdk_restricted_security_j9_0_DISABLED
18:55:07  Disabled Reason:
18:55:07  only applicable for Semeru atm
18:55:07  jdk_restricted_security_j9_0 Finish Time: Wed Dec  4 18:55:07 2024 Epoch Time (ms): 1733356507614
```

**Comment by llxia** [comment]
> But the test is disabled due to "only applicable for Semeru atm". Can't this test be used for nightly build or personal build?

As per your request https://github.com/adoptium/aqa-tests/pull/5782#discussion_r1868024958, the test is set to be Semeru-specific. If you wish to run it against OpenJ9, it would no longer be Semeru-specific. Could you please clarify your requirements?

**Comment by taoliult** [comment]
> > But the test is disabled due to "only applicable for Semeru atm". Can't this test be used for nightly build or personal build?
> 
> As per your request [#5782 (comment)](https://github.com/adoptium/aqa-tests/pull/5782#discussion_r1868024958), the test is set to be Semeru-specific. If you wish to run it against OpenJ9, it would no longer be Semeru-specific. Could you please clarify your requirements?

@llxia Sorry I misunderstood that question about "`Semeru specific`". We would like this tests can be run, for example, same as the `jdk_security` tests, which can be run on Semeru, Nightly and Personal build.

File: openjdk/playlist.xml [hunk 3]
```diff
@@ -1657,6 +1657,42 @@
...
+		<disables>
+			<disable>
+				<comment>only applicable for Semeru atm</comment>
+				<vendor>eclipse</vendor>
+			</disable>
+			<disable>
+				<comment>Not applicable on zos atm</comment>
+				<platform>.*zos.*</platform>
+			</disable>
+		</disables>
```
Comment (llxia) [inline openjdk/playlist.xml:1671]: according to https://github.com/adoptium/aqa-tests/pull/5782#issuecomment-2520549828, please remove all disables. And also remove line 1693 `<impl>ibm</impl>`

**Comment by llxia** [comment]
re for JDK8 test failures, is there a particular reason to have  `adds/jdk` folders?

**Comment by taoliult** [comment]
> re for JDK8 test failures, is there a particular reason to have `adds/jdk` folders?

No, no particular reason, I put those codes there, just for under the same directory with the `src` folder.

```
├── adds
│   └── jdk
│       ├── src
│       └── test
```

**Comment by llxia** [comment]
Ok, it may be better for @LongyuZhang to do the change as he is testing the PR, just in case there is any other structure-related change. @LongyuZhang could you please remove `adds/jdk` in the openjdk8 extension repo? Thanks

**Comment by LongyuZhang** [comment]
Discussed with Lan, will enable this test for JDK11+ currently, with disable aix, mac, and win (track in runtimes/automation/issues/151). 

Test Links (hyc_grinder):
xlinux: 45446
zlinux: 45458
plinux: 45459
aix: (disabled so skip) 45574

**Review by llxia — APPROVED** [review]
LGTM. We will enable JDK8 in a separate PR.

**Review by JasonFengJ9 — APPROVED** [review]

**Comment by llxia** [comment]
@taoliult @jasonkatonica do we need to run these restricted security tests for the 0.49 release (Jan release)?

**Comment by jasonkatonica** [comment]
Hi @llxia we would like to run these tests as soon since we have no coverage for the restricted security mode profile code. This being said we are ok missing the .49 deadlines if the timing does not work for this feature or if we run into issues.

**Comment by taoliult** [comment]
> @taoliult @jasonkatonica do we need to run these restricted security tests for the 0.49 release (Jan release)?

If we can catch up the 0.49 deadline, yes, we can enable these tests for the 0.49 release.

And for the JDK11 `TestProviders` tests failures, I will open an OpenJDK PR for that. The root cause for those failures are the provider SunJSSE fully qualified name are different between JDK11 and JDK17. I will update the SunJSSE for those test cases on JDK11 `TestProviders` tests.

**Comment by llxia** [comment]
Thanks @taoliult and @jasonkatonica 

@LongyuZhang please port this AQA change into https://github.com/adoptium/aqa-tests/tree/v1.0.5-release
The related extension repo changes are in 0.49 already:
https://github.com/ibmruntimes/openj9-openjdk-jdk11/tree/v0.49.0-release/closed/test/jdk
https://github.com/ibmruntimes/openj9-openjdk-jdk17/tree/v0.49.0-release/closed/test/jdk
https://github.com/ibmruntimes/openj9-openjdk-jdk21/tree/v0.49.0-release/closed/test/jdk
https://github.com/ibmruntimes/openj9-openjdk-jdk23/tree/v0.49.0-release/closed/test/jdk

**Pull request merged**